moped-editor/src/components/GridTable/Filters.js

```diff
@@ -685,13 +699,15 @@ const Filters = ({
                       />
                     ) : (
                       <TextField
+                        error={isInvalidInput}
+                        helperText={isInvalidInput ? "Invalid input" : ""}
                         key={`filter-search-value-${filterId}`}
                         id={`filter-search-value-${filterId}`}
                         disabled={!filterParameters[filterId].operator}
                         type={
-                          filterParameters[filterId].type
+                          filterParameters[filterId].type === "date"
                             ? filterParameters[filterId].type
-                            : "text"
+                            : null
```

Comment:
"number" type being used here was causing me issues bc it will automatically set the value to an empty string "" if non ints are in the input and the onChange call wont fire, making it so that i cant check if the input is invalid. [MUI docs](https://github.com/mui/material-ui/blob/master/docs/data/material/components/text-fields/text-fields.md#typenumber) actually recommend avoiding using `type={"number")` for this component and we really only need this distinction here for the "date" type so thats why i modified this

Reply:
thanks for catching this. totally agree that the "number" type input only creates headaches. not critical, but it might be cool to test out using the `inputmode` and `pattern` attributes. GOV.UK did a great [write up](https://technology.blog.gov.uk/2020/02/24/why-the-gov-uk-design-system-team-changed-the-input-type-for-numbers/) about this!

Reply:
@roseeichelmann I re-read John's comment here, and I realized that we did something similar [in this currency component](https://github.com/cityofaustin/atd-moped/blob/main/moped-editor/src/views/projects/projectView/DollarAmountIntegerField.js). `inputmode` would be a good addition here, but it would also add more logic of when/when not to add that attribute based on the filter `type`. 

Feels like future work to handle different `type`s by composing input components instead of toggling prop values like we do now. Curious what others think!

Reply:
yep. this also makes me appreciate the [Yup-style validation](https://github.com/cityofaustin/atd-moped/blob/f7e7a5f1c0106d1ffecd289eaa5a0c9b6f281b30/moped-editor/src/views/projects/projectView/ProjectWorkActivity/utils/form.js#L22) where you can pass in a message per validation rule. but it definitely makes sense to hold off until we see a bigger need for validation on these filters.